WcaOnRails/app/controllers/api/v0/competitions_controller.rb

```diff
@@ -1,11 +1,55 @@
 # frozen_string_literal: true
 class Api::V0::CompetitionsController < Api::V0::ApiController
-  def show
-    competition = Competition.where(id: params[:id], showAtAll: true).first
-    unless competition
-      render json: { error: "Competition with id #{params[:id]} not found" }, status: 404
-      return
+  def index
+    managed_by_user = nil
+    if params[:managed_by_me].present?
+      require_scope!("manage_competitions")
+      managed_by_user = current_api_user
     end
+
+    competitions = Competition.search(params[:q], params: params, managed_by_user: managed_by_user)
+    competitions = competitions.includes(:delegates, :organizers)
+
+    paginate json: competitions
+  end
+
+  def show
+    competition = competition_from_params
     render json: competition
   end
+
+  def show_wcif
+    competition = competition_from_params
+    require_can_manage!(competition)
+
+    render json: competition.wcif.json
+  end
+
+  private def competition_from_params
+    id = params[:competition_id] || params[:id]
+    competition = Competition.find_by_id(id)
+
+    # If this competition exists, but is not publicly visible, then only show it
+    # to the user if they are able to manage the competition.
+    if competition && !competition.showAtAll && !can_manage?(competition)
+      competition = nil
+    end
+
+    raise WcaExceptions::NotFound.new("Competition with id #{id} not found") unless competition
+    competition
+  end
+
+  private def can_manage?(competition)
+    current_api_user && doorkeeper_token.scopes.exists?("manage_competitions") && current_api_user.can_manage_competition?(competition)
```

Comment:
This should be equivalent, but possibly bad in other ways:

`current_api_user&.can_manage_competition?(competition) && doorkeeper_token.scopes.exists?("manage_competitions")`